### PR TITLE
Fix expanding variables in custom section

### PR DESCRIPTION
**Pull request by koterpillar**
Serverless [supports variables](https://serverless.com/framework/docs/providers/aws/guide/variables/) in the configuration file.

However, they are not expanded in the plugin _constructor_ yet (https://forum.serverless.com/t/serverless-plugin-resolving-variable-references/2233/2), so specifying `docker: false` via a _variable_ does not turn off Docker.

This PR fixes the expanding by only checking the value of `docker` _after_ the constructor.

See `fix-variables-test` for a failing test.

### CODE REVIEW

File: integration-test/skeleton/serverless.yml [hunk 1]
```diff
@@ -33,4 +33,9 @@ custom:
       - one
       - two
       - --three
-    docker: DOCKER
+    # environment variables appear as strings. Index that string into a
+    # dictionary with actual boolean values to get the boolean back
+    docker: ${self:custom.boolean.${env:FORCE_DOCKER, "DOCKER_DEFAULT"}}
+  boolean:
```
Comment (dmoverton) [inline integration-test/skeleton/serverless.yml]: why is this necessary?

Comment (dmoverton) [reply integration-test/skeleton/serverless.yml]: Or do I really not want to know?

Comment (koterpillar) [reply integration-test/skeleton/serverless.yml]: There's the comment above: environment variables are strings, so if you say

```yaml
docker: ${env:FORCE_DOCKER}
```

it'll expand to:

```yaml
docker: "false"
```

However, you can use `self:custom` to get a value _out of your file_, and indirection is allowed, so `self:custom.boolean.true` expands to a _boolean_ value `true`.
